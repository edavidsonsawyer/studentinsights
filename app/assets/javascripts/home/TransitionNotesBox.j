import React from 'react';
import PropTypes from 'prop-types';
import Card from '../components/Card';


// If it's the benchmark assessment window, show the dialog box.
// This only works for the first usage scenario.
const params = {
  benchmarkPeriodText: 'Winter 2018',
  url: '/schools/hea/reading/3/entry'
};
export default function ReadingDataEntryBox(props) {
  const {educatorLabels, style, titleStyle} = props;  
  if (educatorLabels.indexOf('enable_reading_benchmark_data_entry') === -1) return null;
  
  const {benchmarkPeriodText, url} = params;
  return (
    <div style={style}>
      <div style={titleStyle}>Reading benchmarks: {benchmarkPeriodText}</div>
      <Card style={{border: 'none'}}>
        <div>Add <a style={{fontWeight: 'bold'}} href={url}>benchmark reading data</a> for your students, and you'll have new tools for making reading groups and visualizing student growth over time.</div>
      </Card>
    </div>
  );
}
ReadingDataEntryBox.propTypes = {
  educatorLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
  style: PropTypes.object,
  titleStyle: PropTypes.object
};