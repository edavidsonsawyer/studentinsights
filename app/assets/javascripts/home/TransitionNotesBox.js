import React from 'react';
import PropTypes from 'prop-types';
import Card from '../components/Card';
import {toSchoolYear} from '../helpers/schoolYear';

export default class TransitionNotesBox extends React.Component {
  render() {
    const {nowFn} = this.context;
    const {style, titleStyle, educatorLabels} = this.props;
    const showTransitionNotes = (educatorLabels.indexOf('enable_transition_note_features') !== -1);
    if (!showTransitionNotes) return null;

    const schoolYear = toSchoolYear(nowFn());
    return (
      <div style={style}>
        <div style={titleStyle}>School transitions for {schoolYear+1}</div>
        <Card style={{border: 'none'}}>
          <div>Add <a style={{fontWeight: 'bold'}} href="/counselors/transitions">transition notes</a> to introduce your colleagues to the students you worked with this year.</div>
        </Card>
      </div>
    );
  }
}
TransitionNotesBox.contextTypes= {
  nowFn: PropTypes.func.isRequired
};
TransitionNotesBox.propTypes = {
  educatorLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
  style: PropTypes.object,
  titleStyle: PropTypes.object
};