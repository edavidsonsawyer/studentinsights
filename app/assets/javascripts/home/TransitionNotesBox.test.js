import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withDefaultNowContext} from '../testing/NowContainer';
import TransitionNotesBox from './TransitionNotesBox';


function renderIntoEl(element) {
  const el = document.createElement('div');
  ReactDOM.render(element, el);
  return el;
}

function testProps(props) {
  return {
    educatorLabels: [],
    style: {
      color: 'red'
    },
    titleStyle: {
      color: 'purple'
    },
    ...props
  };
}

function testEl(props) {
  return withDefaultNowContext(<TransitionNotesBox {...props} />);
}

it('renders without crashing', () => {
  const props = testProps();
  renderIntoEl(testEl(props));
});

it('renders nothing if label not set', () => {
  const props = testProps();
  const el = renderIntoEl(testEl(props));
  expect($(el).text()).toEqual('');
});

it('snapshots if label is set', () => {
  const props = testProps({
    educatorLabels: ['enable_transition_note_features']
  });
  const tree = renderer
    .create(testEl(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
