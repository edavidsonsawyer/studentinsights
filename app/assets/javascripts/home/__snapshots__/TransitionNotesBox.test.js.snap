// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots if label is set 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  <div
    style={
      Object {
        "color": "purple",
      }
    }
  >
    School transitions for 
    2018
  </div>
  <div
    className="Card "
    style={
      Object {
        "border": "none",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div>
      Add 
      <a
        href="/counselors/transitions"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        transition notes
      </a>
       to introduce your colleagues to the students you worked with this year.
    </div>
  </div>
</div>
`;
